core/arch/arm/kernel/generic_boot.c

```diff
@@ -927,6 +931,12 @@ static void init_external_dt(unsigned long phys_dt)
 	IMSG("Non-secure external DT found");
 }
 
+static int mark_tzdram_as_reserved(struct dt_descriptor *dt)
+{
+	return add_res_mem_dt_node(dt, "optee_core", CFG_TZDRAM_START,
+				   CFG_TZDRAM_SIZE);
+}
+
```

Comment:
Do all platforms want to share this information with the non-secure world?

Reply:
I don't think there is a way around it, alternatively we would need to manipulate the memory node. This however will only really work if OP-TEE is at the end or beginning of RAM. Otherwise the memory node needs to be split and the kernel needs to deal with non-continuous memory.
Either way, if an adversary is controlling the normal world and has access to the MMU mappings, it's not hard to figure out where the memory holes are, IMO we don't gain anything by trying to hide the OP-TEE location from the normal world.

Reply:
I don't see how it can be a problem. On the contrary, we don't want the kernel to create mappings for the secure memory.

Reply:
ok